src/Foundatio.Storage.SshNet/Storage/SshNetFileStorage.cs

```diff
@@ -13,384 +14,432 @@
-                pagingLimit++;
-
-            var list = await GetFileListAsync(searchPattern, pagingLimit, skip, cancellationToken).AnyContext();
-            bool hasMore = false;
-            if (list.Count == pagingLimit) {
-                hasMore = true;
-                list.RemoveAt(pagingLimit - 1);
-            }
+    private async Task<List<FileSpec>> GetFileListAsync(string searchPattern = null, int? limit = null, int? skip = null, CancellationToken cancellationToken = default) {
+        using var _ = _logger.BeginScope(s => s.Property("SearchPattern", searchPattern).Property("Limit", limit).Property("Skip", skip));
+        
+        if (limit is <= 0)
+            return new List<FileSpec>();
 
-            return new NextPageResult {
-                Success = true,
-                HasMore = hasMore,
-                Files = list,
-                NextPageFunc = hasMore ? r => GetFiles(searchPattern, page + 1, pageSize, cancellationToken) : null
-            };
-        }
+        var list = new List<FileSpec>();
+        var criteria = GetRequestCriteria(searchPattern);
 
-        private async Task<List<FileSpec>> GetFileListAsync(string searchPattern = null, int? limit = null, int? skip = null, CancellationToken cancellationToken = default) {
-            if (limit is <= 0)
-                return new List<FileSpec>();
+        EnsureClientConnected();
 
-            var list = new List<FileSpec>();
-            var criteria = GetRequestCriteria(searchPattern);
+        // NOTE: This could be very expensive the larger the directory structure you have as we aren't efficiently doing paging.
+        int? recordsToReturn = limit.HasValue ? skip.GetValueOrDefault() * limit + limit : null;
+        await GetFileListRecursivelyAsync(criteria.Prefix, criteria.Pattern, list, recordsToReturn, cancellationToken).AnyContext();
 
-            EnsureClientConnected();
+        if (skip.HasValue)
+            list = list.Skip(skip.Value).ToList();
 
-            // NOTE: This could be very expensive the larger the directory structure you have as we aren't efficiently doing paging.
-            int? recordsToReturn = limit.HasValue ? skip.GetValueOrDefault() * limit + limit : null;
-            await GetFileListRecursivelyAsync(criteria.Prefix, criteria.Pattern, list, recordsToReturn, cancellationToken).AnyContext();
+        if (limit.HasValue)
+            list = list.Take(limit.Value).ToList();
 
-            if (skip.HasValue)
-                list = list.Skip(skip.Value).ToList();
+        return list;
+    }
 
-            if (limit.HasValue)
-                list = list.Take(limit.Value).ToList();
+    private async Task GetFileListRecursivelyAsync(string prefix, Regex pattern, ICollection<FileSpec> list, int? recordsToReturn = null, CancellationToken cancellationToken = default) {
+        using var _ = _logger.BeginScope(s => s.Property("Prefix", prefix).Property("Pattern", pattern.ToString()));
```

Comment:
Why are you starting a logging scope here? You only do logging scopes if you want every single log message inside of the scope to all have the same properties. I don't see any benefit to having prefix and pattern on every single message. You can log out the prefix and pattern at the start of the operation.

Reply:
because I want to see why it was picked up if I'm looking at a specific log message for the file.

Reply:
You can see that from the log message at the start of the operation. I just don't see any value in adding those properties to every single log message.

Reply:
Fixed